syft/file/cataloger/executable/cataloger.go

```diff
@@ -0,0 +1,234 @@
+package executable
+
+import (
+	"bytes"
+	"debug/elf"
+	"debug/macho"
+	"encoding/binary"
+	"fmt"
+
+	"github.com/bmatcuk/doublestar/v4"
+	"github.com/dustin/go-humanize"
+
+	"github.com/anchore/syft/internal"
+	"github.com/anchore/syft/internal/bus"
+	"github.com/anchore/syft/internal/log"
+	"github.com/anchore/syft/syft/event/monitor"
+	"github.com/anchore/syft/syft/file"
+	"github.com/anchore/syft/syft/internal/unionreader"
+)
+
+type Config struct {
+	MIMETypes []string `json:"mime-types" yaml:"mime-types" mapstructure:"mime-types"`
+	Globs     []string `json:"globs" yaml:"globs" mapstructure:"globs"`
+}
+
+type Cataloger struct {
+	config Config
+}
+
+func DefaultConfig() Config {
+	return Config{
+		MIMETypes: internal.ExecutableMIMETypeSet.List(),
+		Globs:     nil,
+	}
+}
+
+func NewCataloger(cfg Config) *Cataloger {
+	return &Cataloger{
+		config: cfg,
+	}
+}
+
+func (i *Cataloger) Catalog(resolver file.Resolver) (map[file.Coordinates]file.Executable, error) {
+	locs, err := resolver.FilesByMIMEType(i.config.MIMETypes...)
+	if err != nil {
+		return nil, fmt.Errorf("unable to get file locations for binaries: %w", err)
+	}
+
+	locs, err = filterByGlobs(locs, i.config.Globs)
+	if err != nil {
+		return nil, err
+	}
+
+	prog := catalogingProgress(int64(len(locs)))
```

Comment:
This might not be for this PR but for the general syft v1.0 work. I noticed that this, the fileDigest, and the fileContent, are all catalogers where we invoke or have some concept of catalogingProgress. Is this something we're trying to remove from the library code and only surface in the cmd package or is it ok that this concern is still in a handful of cataloger?

https://github.com/search?q=repo%3Aanchore%2Fsyft+catalogingProgress&type=code

Reply:
they all have subtly different implementations so it isn't worth trying to share the implementation / export. (But yeah, it's also not an external concern even if they were the same.)